cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -71,11 +94,49 @@ public HBaseMetricsTable(DatasetContext datasetContext, DatasetSpecification spe
     this.columnFamily = TableProperties.getColumnFamilyBytes(spec.getProperties());
   }
 
+  private void initializeV3Vars(DatasetSpecification spec) {
+    boolean isV3Table = spec.getName().contains("v3");
+    this.scanExecutor = null;
+    this.rowKeyDistributor = null;
+    if (isV3Table) {
+      RejectedExecutionHandler callerRunsPolicy = new RejectedExecutionHandler() {
+
+        @Override
+        public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+          REJECTION_LOG.info(
+            "No more threads in the HBase scan thread pool. Consider increase {}. Scan from caller thread {}",
```

Comment:
Oh wait, you are trying to say: because the thread pool is exhausted, the scan is performed synchronously in the  caller thread. Then perhaps phrase it as "Performing scan in caller thread {}".